### PR TITLE
USB: Add Popegg (MPR-G600) printer emulation

### PR DESCRIPTION
**Pull request by Florin9doi**
### Description of Changes
Popegg (MPR-G600A) printer emulation.

Currently, the print job is written to a file which should be then converted to a PPM image using this program: https://github.com/echiu64/gutenprint/blob/master/test/pixma_parse.c
It can be improved by including pixma_parse as submodule or in a 3rdparty folder.

### Suggested Testing Steps
* Compatibility: https://gist.github.com/Florin9doi/57d496d3790d30704b596eb9c9c50205

### CODE REVIEW

File: pcsx2/USB/usb-printer/usb-printer.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
+#include "PrecompiledHeader.h"
```
Comment (Mrlinkwii) [inline pcsx2/USB/usb-printer/usb-printer.cpp:1]: You're missing the copyright header :slightly_smiling_face:

File: pcsx2/USB/usb-printer/usb-printer.cpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+		const uint8_t ep_type = p->ep->type;
+
+		switch (p->pid)
+		{
+			case USB_TOKEN_OUT:
+				usb_packet_copy(p, data, MIN(p->iov.size, sizeof(data)));
+				if (data[0] == 0x1b && data[1] == 0x5b && data[2] == 0x4b && data[3] == 0x1c)
+				{
+					s->cmd_state = 1;
+				} else if (s->print_file < 0
```
[the remark this answers is not in the document]
Comment (Florin9doi) [reply pcsx2/USB/usb-printer/usb-printer.cpp:99]:
```suggestion

				} else if (s->print_file < 0
```

[the remark this answers is not in the document]
Comment (lightningterror) [reply pcsx2/USB/usb-printer/usb-printer.cpp]: else if should be below the bracket.

**Comment by lightningterror** [comment]
Commit can be better named, adding description to the commit too.

**Comment by refractionpcsx2** [comment]
What's the status of this PR?

**Comment by Florin9doi** [comment]
It works, at least for PrintFan game, but the experience isn't ideal. Any feedback about pixma_parse integration ? Or it can be included later

**Comment by refractionpcsx2** [comment]
I'd prefer avoiding using 3rd party stuff if we can help if, but if it can't be helped then yeah, include it in its own folder in 3rd party.

File: pcsx2/USB/usb-printer/usb-printer.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+				{
+					const unsigned int len = sprintf((char*)data + 2, "xxDWS:NO;DOC:4,00,NO;DSC:NO;DBS:NO;DJS:NO;");
+					data[0] = 0x00;
+					data[1] = 1 + len;
+					actual_length = 2 + len;
+					s->cmd_state = 0;
+				}
+				else
+				{
+					const unsigned char status[] = {0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xff, 0x00, 0x00, 0x00, 0xff};
```
Comment (jackun) [inline pcsx2/USB/usb-printer/usb-printer.cpp:126]: Would be nice to have a some info what these magic numbers are and links etc for reference maybe. google-fu sucks these days

**Comment by Florin9doi** [comment]
On hold until #5184 is merged

**Comment by refractionpcsx2** [comment]
Is there any chance we can get it to save in a better format than raw? like BMP or something? trying to open it photoshop has no clue what the format is or the size or anything, it's not very user friendly.

**Comment by refractionpcsx2** [comment]
Why did you close this?

**Comment by MMatiasMG** [comment]
Why this was closed?

**Pull request closed without merging**